composers/getCountriesComposer.ts

```diff
@@ -14,11 +14,7 @@ export class GetCountriesComposer extends BaseProcessor {
                 const db = this.executionContext.getSharedResource('dataProvider')
                 const connection = await db.getConnection()
                 if (!connection) {
-                    return reject({
-                        successful: false,
-                        message: 'Invalid connection',
-                        httpStatus: 500
-                    })
+                    return reject(this.handleError({message: `Invalid connection`}, `getCountriesComposer.fx`, 500))
```

Comment:
There may not be, but is there a way in TypeScript to get the current method/class dynamically for use instead of a hardcoded string?  (Probably not a big deal here, but was wondering in general.)

Reply:
In a short answer: Yes and No =P
You can reference `constructor.name` and get a class name BUT it's not a good practice because code could be minified/uglified and the name would change (since it is just a string)

I think it would be a good idea to create a className variable like you did in the converters and use that instead. A variable would work perfectly and could replace all the magic strings.

I will create a separate work item to do that work though

Reply:
Would this work? this.constructor.toString().match(/\w+/g)[1];  found on StackOverflow

Reply:
it would but if we ever minified the code or uglified it, this would no longer work. The process of minification works like this:

````
export class MyClass {
  constructor() {}
  doIt() {
    console.log('1 + 2 = ' + Summer.sum(1, 2));
  }
}
class Summer {
  public static sum(a: number, b: number) {
    return a + b;
  }
}

// BECOMES
export class a {
  constructor(){}
  b() {
    console.log('1 + 2 = ' + c.d(1, 2));
  }
}
class c {
  c.prototype.d = function(e, f) { 
    return e + f;
  }
}
````
So the code unminified would return `MyClass` and `Summer` for the constructors BUT the minified version would return `a` and `c`